1.fee_bumping/fee_bumping.md

```diff
@@ -153,25 +153,208 @@ Bitcoin blockchain, such as exchanges or custodians:
-
-TODO
-
-## Overview of what RBF is
-
-TODO
-
-## History of RBF
-
-TODO
+# Transaction replacement
+
+The concept of being able to 'replace' an unconfirmed transaction has been
+around since the very earliest days of Bitcoin. Early versions of the
+Bitcoin reference implementation [allowed transactions to be replaced][full_replacement] by
+constructing and broadcasting a replacement transaction with modified
+`nSequence` numbers in the transaction inputs.
+
+Satoshi understood the need for transaction fees to increase as the subsidy
+decreased, saying in the [whitepaper][] 'Once a predetermined number of coins
+have entered circulation, the incentive can transition entirely to transaction
+fees and be completely inflation free.' He also understood that users would
+need a way to be able to bump the fee on a transaction if it got stuck. There
+was [a question on the bitcointalk forum][bct q] in 2010:
+
+> Just wondering about the following example :
+> I broadcast a transaction, sending X coins to some address.
+> Doesn't get included in blocks for a while because I don't include a fee.
+> 
+> Do I have a way to cancel it and broadcast it again with a fee this time ?
+
+Satoshi [replied][bct a]:
+
+> There's a possible design for far in the future:
+> 
+> You intentionally write a double-spend.  You write it with the same inputs
+> and outputs, but this time with a fee.  When your double-spend gets into a
+> block, the first spend becomes invalid.  [...]
+> 
+> It's easier said than implemented.  There would be a fair amount of work[...]
+
+There are two main areas of difficulty that Satoshi is talking about: wallet
+design and UX, and node and miner policy.
+
+The wallet design and UX questions (how should a wallet track and display
+conflicting transactions? How should it present a user interface for bumping
+fee?) can be resolved by individual wallets and applications.
+
+The node an miner policy difficulties require some kind of shared
+understanding of how other nodes and miners on the network will behave. There
+are two major difficulties to consider:
+
+1. **DoS** - how do we prevent malicious actors from consuming node resources
+   by repeatedly broadcasting replacement transactions for zero cost?
+2. **Incentive alignment** - how do we make sure that replacement transactions
+   are incentive-compatible and miners want to mine them?
+
+Satoshi's first version of transaction replacement failed on both of these
+counts. The `nSequence` number could be incremented millions of times without
+changing the fee, forcing each node in the network to re-validate all of the
+transactions many times at zero cost to the attacker. The replacement logic also
+didn't consider the transaction fee at all, and would replace an old
+transaction with a new transaction even if the replacement had a lower fee. If
+a miner was running this software, the replacement behaviour would be costing
+him money since he'd be mining transactions with lower fees.
```

Comment:
> Dorky nitpick.

Keep 'em coming. There's more wisdom in a Harding dorky nitpick than months of cryptotwitter :)

Updated to:

```
Satoshi's first version of transaction replacement failed on both of these
counts. The `nSequence` number could be incremented millions of times without
changing the fee, wasting bandwidth and compute resources on all nodes in the
network. The replacement logic also didn't require the replacement transaction
to increase the fee, and could replace an old transaction with a new
transaction even if the replacement had a lower fee. If a miner was running
this software, the replacement behaviour would be costing him money since he'd
be mining transactions with lower fees.
```